glean-core/ios/GleanTests/GleanTests.swift

```diff
@@ -328,6 +329,32 @@ class GleanTests: XCTestCase {
         }
     }
 
+    func testSettingSourceTagsBeforeInitialization() {
+        // This test relies on Glean not being initialized
+        Glean.shared.testDestroyGleanHandle()
+
+        XCTAssert(Glean.shared.setSourceTags(["valid-tag", "tag-valid"]))
+
+        // Set the last time the "metrics" ping was sent to now. This is required for us to not
+        // send a metrics pings the first time we initialize Glean and to keep it from interfering
+        // with these tests.
+        let now = Date()
+        Glean.shared.metricsPingScheduler.updateSentDate(now)
+        // Restart glean
+        Glean.shared.resetGlean(clearStores: false)
+
+        let host = URL(string: Configuration.Constants.defaultTelemetryEndpoint)!.host!
+        stub(condition: isHost(host)) { data in
```

Comment:
hmpf, eventually we should extend `stubServerReceive` to have the headers.
Fine for now here though.